### PR TITLE
fix: Only consider Wreck Materials to have an image is the image property is a string MCABAND-285

### PR DESCRIPTION
**Pull request by WillFortMadeTech**
This changes aims to ensure that a Wreck Material cannot be submitted without having an image, which we hope will ensure that the user sees an error they can address before they submit their Wreck Report.

### CODE REVIEW

File: webapp/api/routes/report/check-your-answers.test.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    });
+    fs.promises.readFile.mockResolvedValue('base64encodeddata');
+
+    await agent.post('/report/confirmation').send({ 'property-declaration': 'on' });
+
+    const wreckMaterial = axios.post.mock.calls[1][1];
+    expect(wreckMaterial.value).toBeNull();
+  });
+
+  it('still submits the wreck material when the image file cannot be read', async () => {
```
Comment (willgibson-madetech) [inline webapp/api/routes/report/check-your-answers.test.js]: @WillFortMadeTech I started adding tests for the existing `check-your-answers.js` behaviours and it's thrown this up.

I don't think we want this behaviour, but I will finish the as is coverage then loop back to it.

Comment (WillFortMadeTech) [reply webapp/api/routes/report/check-your-answers.test.js]: I have amended the existing test and functionality for this to return the user back to check-your-answers if the image is undefined.

File: webapp/api/routes/report/check-your-answers.test.js [hunk 2]
```diff
@@ -195,19 +195,18 @@ describe('POST /report/confirmation — image file reading', () => {
...
 
-    // SubmitWreckMaterial is still called even though readFile failed
+    expect(res.status).toBe(302);
+    expect(res.headers.location).toBe('/report/check-your-answers');
     const wreckMaterialCall = axios.post.mock.calls[1];
-    expect(wreckMaterialCall).toBeDefined();
-    // image is left as the original filename string, not replaced with an object
-    expect(typeof wreckMaterialCall[1].image).toBe('string');
+    // expect image to be undefined
+    expect(wreckMaterialCall).toBeUndefined();
```
Comment (willgibson-madetech) [inline webapp/api/routes/report/check-your-answers.test.js:208]: What does the user see? I think they should get a decent, helpful error message, which we coud add an assertion for here.

Comment (willgibson-madetech) [reply webapp/api/routes/report/check-your-answers.test.js:208]: It is a bit of an edge case though. Maybe we could just log something that enables us to track this and review in a the future to see how often it happens?

File: webapp/api/routes/report/check-your-answers.test.js [hunk 3]
```diff
@@ -195,18 +195,16 @@ describe('POST /report/confirmation — image file reading', () => {
...
+    const res = await agent.post('/report/confirmation').send({ 'property-declaration': 'on' });
 
-    // SubmitWreckMaterial is still called even though readFile failed
-    const wreckMaterialCall = axios.post.mock.calls[1];
-    expect(wreckMaterialCall).toBeDefined();
-    // image is left as the original filename string, not replaced with an object
-    expect(typeof wreckMaterialCall[1].image).toBe('string');
+    expect(res.status).toBe(200);
+    expect(res.text).toContain('Upload failed, please re-upload your image');
+    expect(axios.post.mock.calls[1]).toBeUndefined();
```
Comment (willgibson-madetech) [inline webapp/api/routes/report/check-your-answers.test.js]: I think it's still worth having `wreckMaterialCall` for grokability...


```suggestion
    const wreckMaterialCall = axios.post.mock.calls[1];
    expect(wreckMaterialCall).toBeUndefined();
```

**Review by willgibson-madetech — APPROVED** [review]
